static/js/components/dashboard/FinalExamCard.js

```diff
@@ -137,63 +78,156 @@ const schedulableCourseList = R.compose(
   R.defaultTo({}),
 );
 
-const schedulableCard = (profile, program, navigateToProfile, submitPearsonSSO, pearson) => cardWrapper(
-  accountCreated(profile, navigateToProfile),
-  <div key="schedulable" className="exam-scheduling">
-    <SpinnerButton
-      className="mdl-button exam-button"
-      component={Button}
-      spinning={isProcessing(pearson)}
-      onClick={submitPearsonSSO}
-      ignoreRecentlyClicked={true}
-    >
-      Schedule an exam
-    </SpinnerButton>
-    <div className="program-info">
-      You are ready to schedule an exam for:
-      <ul>
-        { schedulableCourseList(program) }
-      </ul>
-    </div>
-  </div>,
-  errorDisplay(pearson)
-);
-
 type Props = {
   profile:            Profile,
   program:            Program,
   navigateToProfile:  () => void,
   submitPearsonSSO:   () => void,
   pearson:            PearsonAPIState,
+  ui:                 UIState,
+  showToPearsonSiteDialog: (open: boolean) => void,
 };
 
 export default class FinalExamCard extends React.Component<void, Props, void> {
```

Comment:
is there a reason you need to refactor the separate functions into the class? if not, I'd prefer to keep the implementation as-is.